src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs

```diff
@@ -0,0 +1,188 @@
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.FindUsages
+{
+    internal static class RichNavOptions
+    {
+        internal const string RichNavAvailableOptionName = "RichNavAvailable";
+    }
+
+    [Export(typeof(IScopeFilterFactory))]
+    [TableManagerIdentifier("FindAllReferences*")]
+    [TableManagerIdentifier("FindResults*")]
+    [Replaces(PredefinedScopeFilterNames.EntireSolutionScopeFilter)]
+    [Replaces(PredefinedScopeFilterNames.AllItemsScopeFilter)]
+    [DeferCreation(OptionName = RichNavOptions.RichNavAvailableOptionName)] // This factory will not be loaded unless this option is set to Boolean true
+    [DefaultScope]
+    [Name(PredefinedScopeFilterNames.LoadedSolutionScopeFilter)]
+    [Order(Before = PredefinedScopeFilterNames.AllItemsScopeFilter)]
+    internal class LoadedSolutionScopeFilterFactory : IReplacingScopeFilterFactory
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public LoadedSolutionScopeFilterFactory()
+        {
+        }
+
+        public IErrorListFilterHandler CreateFilter(IWpfTableControl tableControl)
+        {
+            // We're only replacing an existing filter, and not creating a new one.
+            return null;
+        }
+
+        public IErrorListFilterHandler ReplaceFilter(IWpfTableControl tableControl, string filterIdentifier)
+        {
+            if (filterIdentifier == PredefinedScopeFilterNames.AllItemsScopeFilter)
+            {
+                return new LoadedSolutionFilterHandler(ServicesVSResources.Loaded_items);
+            }
+            else if (filterIdentifier == PredefinedScopeFilterNames.EntireSolutionScopeFilter)
+            {
+                return new LoadedSolutionFilterHandler(ServicesVSResources.Loaded_solution);
+            }
+
+            return null; // Don't replace
+        }
+    }
+
+    internal abstract class RoslynFilterHandler : FilterHandlerBase
+    {
+        private readonly int _id;
+        private readonly string _displayName;
+        private readonly ItemOrigin _origin;
+
+        protected RoslynFilterHandler(int id, string displayName, ItemOrigin origin)
+        {
+            _id = id;
+            _displayName = displayName;
+            _origin = origin;
+        }
+
+        public sealed override int FilterId => _id;
+
+        public sealed override string FilterDisplayName => _displayName;
+
+        public sealed override IEntryFilter GetFilter(out string displayText)
+        {
+            displayText = _displayName;
+            return new ItemOriginFilter(_origin);
+        }
+    }
+
+    internal class LoadedSolutionFilterHandler : RoslynFilterHandler
+    {
+        private const int LoadedSolutionFilterHandlerFilterId = 20;
+        public LoadedSolutionFilterHandler(string displayName)
+            : base(LoadedSolutionFilterHandlerFilterId, displayName, ItemOrigin.ExactMetadata)
```

Comment:
does this match Exact and ExactMetadata @AmadeusW ?

Reply:
Yes

```
    public enum ItemOrigin
    {
        Exact = 0,
        ExactMetadata = 1000,
        Indexed = 2000,
        IndexedInRepo = 2100,
        IndexedInOrganization = 2200,
        IndexedInThirdParty = 2300,
        Other = int.MaxValue,
    }
```